recOrder/plugin/main_widget.py

```diff
@@ -2014,6 +2027,51 @@ def acq_ret_ori_phase(self):
         # Start Thread
         self.worker.start()
 
+    def _dump_gui_state(self, save_dir: StrOrBytesPath):
+        """Collect and save the current GUI settings to a YAML file.
+
+        Parameters
+        ----------
+        save_dir : str | bytes | PathLike[str] | PathLike[bytes]
+            directory to save
+        """
+        gui_state = {
+            "Run Calibration": {
+                "Swing": self.swing,
+                "Wavelength": self.wavelength,
+                "Illumination Scheme": self.calib_scheme,
+                "Calibration Mode": self.calib_mode,
+                "Config Group": self.config_group,
+            },
+            "Capture Background": {
+                "Background Folder Name": self.bg_folder_name,
+                "Number of Images to Average": self.n_avg,
+            },
+            "Acquisition Settings": {
+                "Z Start": self.z_start,
+                "Z End": self.z_end,
+                "Z Step": self.z_step,
+                "Acquisition Mode": self.acq_mode,
+                "BF Channel": self.ui.cb_acq_channel.itemText(
+                    self.ui.cb_acq_channel.currentIndex()
+                ),
+            },
+            "General Reconstruction Settings": {
+                "Background Correction": self.bg_option,
+                "Background Path": self.current_bg_path,
+                "Wavelength": self.recon_wavelength,
+                "Objective NA": self.obj_na,
+                "Condenser NA": self.cond_na,
+                "Camera Pixel Size": self.ps,
+                "RI of Objective Media": self.n_media,
+                "Magnification": self.mag,
+                "Orientation Offset": self.orientation_offset,
+            },
+        }
+        save_path = os.path.join(save_dir, "gui_state.yml")
+        with open(save_path, "w") as f:
```

Comment:
This is great! This is exactly the format I had in mind. 

Do you mind:
1. Adding the "Phase Reconstruction Settings"? I think there are just "Z Padding", "Regularizer", and "Strength"
2. Figuring out why the order changed on hummingbird (see attached)? I'm guessing it's related to `yaml.dump`?
[gui_state.txt](https://github.com/mehta-lab/recOrder/files/10026859/gui_state.txt)

Otherwise, LGTM!

Reply:
> Adding the "Phase Reconstruction Settings"

I meant to add those as well... slipped over my mind (again!)